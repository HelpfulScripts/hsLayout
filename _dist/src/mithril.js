if (!global['window']) {
    console.log('creating non-browser polyfill');
    global['window'] = require("mithril/test-utils/browserMock.js")();
    global['document'] = window.document;
}
export const m = require("mithril");
const o = require("mithril/ospec/ospec");
o.root = window.document.createElement("div");
export { o };
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoibWl0aHJpbC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uLy4uL3NyYy9taXRocmlsLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUlBLElBQUksQ0FBQyxNQUFNLENBQUMsUUFBUSxDQUFDLEVBQUU7SUFDbkIsT0FBTyxDQUFDLEdBQUcsQ0FBQywrQkFBK0IsQ0FBQyxDQUFDO0lBRTdDLE1BQU0sQ0FBQyxRQUFRLENBQUMsR0FBRyxPQUFPLENBQUMsbUNBQW1DLENBQUMsRUFBRSxDQUFDO0lBQ2xFLE1BQU0sQ0FBQyxVQUFVLENBQUMsR0FBRyxNQUFNLENBQUMsUUFBUSxDQUFDO0NBRXhDO0FBS0QsTUFBTSxDQUFDLE1BQU0sQ0FBQyxHQUFHLE9BQU8sQ0FBQyxTQUFTLENBQUMsQ0FBQztBQU9wQyxNQUFNLENBQUMsR0FBRyxPQUFPLENBQUMscUJBQXFCLENBQUMsQ0FBQztBQUN6QyxDQUFDLENBQUMsSUFBSSxHQUFHLE1BQU0sQ0FBQyxRQUFRLENBQUMsYUFBYSxDQUFDLEtBQUssQ0FBQyxDQUFDO0FBRTlDLE9BQU8sRUFBRSxDQUFDLEVBQUUsQ0FBQyJ9