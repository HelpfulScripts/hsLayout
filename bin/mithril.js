"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.m = require("mithril");
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoibWl0aHJpbC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uL3NyYy9taXRocmlsLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7O0FBUWEsUUFBQSxDQUFDLEdBQUcsT0FBTyxDQUFDLFNBQVMsQ0FBQyxDQUFDIn0=