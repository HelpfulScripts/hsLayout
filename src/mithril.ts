/**
 * translates mithril libraries to an ES6 module and provides some Typescript type shortcuts.
 */


/**
 * import and re-export the mithril m objkect
 */
import _m from "mithril";

export const m = _m;
export type Vnode = _m.Vnode<any, any>;
