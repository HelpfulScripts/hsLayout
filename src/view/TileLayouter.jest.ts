import * as hslayout from "../";
import { o, m } from "../mithril";

let cont: any;

const layout = (tileSizes: string[]) =>
    new Promise(resolve => {
        const config = {
        css: "myRow",
        content: [`First: ${tileSizes[0]}`, "Second: remainder"]
        };
        config['tiles'] = tileSizes;
        m.mount(o.root, { view: () => m(hslayout.Layout, config) });
        cont = o.root.childNodes[0];
        resolve();
    });
describe('dummy test', ()=>{
    test('layout exists', ()=>{
        expect(hslayout).toBeDefined();
    });
});

describe("tiles px", () => {
    const tileSizes = ['150px','100px'];
    beforeAll(() => layout(tileSizes));
    describe("snapshots", () => {
        test("rows should be defined", () => expect(cont).toBeDefined());
        test(`hs-row-layout, [${tileSizes}]`, () => expect(cont).toMatchSnapshot());
    });
});

describe("tiles %", () => {
    const tileSizes = ['30%','25%'];
    beforeAll(() => layout(tileSizes));
    describe("snapshots", () => {
        test("rows should be defined", () => expect(cont).toBeDefined());
        test(`hs-row-layout, [${tileSizes}]`, () => expect(cont).toMatchSnapshot());
    });
});
