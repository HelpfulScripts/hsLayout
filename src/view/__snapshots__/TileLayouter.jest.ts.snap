// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tiles % snapshots hs-row-layout, [30%,25%] 1`] = `
<div
  class="hs-layout  hs-tile-layout myRow"
>
  <div
    class="hs-layout  "
    style="top: 0%; left: 0%; width: 30%; height: 25%;"
  >
    <div
      class="hs-leaf"
    >
      First: 30%
    </div>
  </div>
  <div
    class="hs-layout  "
    style="top: 0%; left: 30%; width: 30%; height: 25%;"
  >
    <div
      class="hs-leaf"
    >
      Second: remainder
    </div>
  </div>
</div>
`;

exports[`tiles px snapshots hs-row-layout, [150px,100px] 1`] = `
<div
  class="hs-layout  hs-tile-layout myRow"
>
  <div
    class="hs-layout  "
    style="top: 0%; left: 0%; width: 150px; height: 100px;"
  >
    <div
      class="hs-leaf"
    >
      First: 150px
    </div>
  </div>
  <div
    class="hs-layout  "
    style="top: 100%; left: 0%; width: 150px; height: 100px;"
  >
    <div
      class="hs-leaf"
    >
      Second: remainder
    </div>
  </div>
</div>
`;
